### ISSUE TITLE
valid-typeof: support 'unknown'

### ISSUE THREAD

**Issue by mgol** [eslint/eslint#911]
Ref jshint/jshint#1425

IEs >=8 recognize an additional `typeof` value: `unknown`, see:
http://stackoverflow.com/a/10982960/369687
Specifically:
"Internet Explorer displays “unknown” when the object in question is on the other side of a COM+ bridge. You may not know this or realize this, but MS’s XMLHTTP object is part of a different COM+ object that implements IUnknown; when you call methods on it, you’re doing so over a COM bridge and not calling native JavaScript.

Basically that’s MS’s answer if you try to test or access something that’s not a true part of the JScript engine."

Sometimes it might be desirable to check for this non-standard value. Besides, it's unlikely to make a typo in another value and get `unknown`, it's too different from the rest.

You might want to consider allowing such a value.
